src/Chapter2.hs

```diff
@@ -728,8 +763,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
-
+smartReplicate []       = []
+smartReplicate (x : xs) = replicate x x ++ smartReplicate xs
```

Comment:
That's a great solution! But you can improve it even further by using standard function `concatMap`